File: src/ToDo/Models/Tarefa.cs
using Microsoft.AspNetCore.Components.Web.Virtualization;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ToDo.Models
{
    public class Tarefa
    {
        [Key]
        [Required]
        public virtual int Id { get; set; }
        [Required]
        public virtual string Titulo { get; set; }
        public virtual string Descricao { get; set; }
        [Required]
        public virtual Status Status { get; set; }
        [Required]
        [JsonConverter(typeof(IsoDateTimeConverter))]
        public virtual DateTime DataDeCriacao { get; set; }
        [JsonConverter(typeof(IsoDateTimeConverter))]
        public virtual DateTime? DataDeEncerramento { get; set; }
        [Required]
        [JsonConverter(typeof(IsoDateTimeConverter))]
        public virtual DateTime DataDeVencimento { get; set; }
        [Required]
        public virtual int UsuarioId { get; set; }
        [ForeignKey("UsuarioId")]
        public virtual Usuario Usuario { get; set; }

        public virtual ICollection<Categoria>? Categorias { get; set; }
    }
}

File: src/todo-front/src/models/Tarefa.ts
import Status from './Status'
import { Categoria } from './Categoria'

export class Tarefa {
  Id?: number;
  Titulo: string;
  Descricao: string;
  Status: number;
  DataDeEncerramento?: Date;
  DataDeVencimento: Date;
  UsuarioId: number;
  CategoriasId: number[];

  constructor(
    titulo: string,
    descricao: string,
    status: number,
    dataDeVencimento: Date,
    usuarioId: number,
    categoriasId: number[] = [],
    Id?: number
  ) {
    this.Titulo = titulo;
    this.Descricao = descricao;
    this.Status = status;
    this.DataDeVencimento = dataDeVencimento;
    this.UsuarioId = usuarioId;
    this.CategoriasId = categoriasId;
    if (Id !== undefined) {
      this.Id = Id; // Define apenas se fornecido
    }
  }
}